api/envoy/extensions/filters/http/grpc_json_reverse_transcoder/v3/transcoder.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.grpc_json_reverse_transcoder.v3;
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_json_reverse_transcoder.v3";
+option java_outer_classname = "TranscoderProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_json_reverse_transcoder/v3;grpc_json_reverse_transcoderv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: gRPC-JSON reverse transcoder]
+// gRPC-JSON reverse transcoder :ref:`configuration overview <config_http_filters_grpc_json_reverse_transcoder>`.
+// [#extension: envoy.filters.http.grpc_json_reverse_transcoder]
+
+// [#next-free-field: 6]
+// GrpcJsonReverseTranscoder is the filter configuration for the gRPC JSON
+// reverse transcoder. The reverse transcoder acts as a bridge between gRPC
+// client and an HTTP/JSON server, converting the gRPC request into HTTP/JSON
+// for the HTTP backend and the HTTP/JSON response back to gRPC for the gRPC
+// client.
+message GrpcJsonReverseTranscoder {
+  oneof descriptor_set {
```

Comment:
usage of `oneof` is discouraged: https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md?plain=1#L63-L91.

I assume this was borrowed from how https://github.com/envoyproxy/envoy/blob/c1798d7f9cd36a27cf59ea390bdbe702aea0be3a/api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto#L113 does it, but I'm thinking it would be better to define two separate fields and an order of precedence to conform to the style guide recommendations.